cstrml/balances/src/tests.rs

```diff
@@ -0,0 +1,691 @@
+				assert_eq!(Balances::total_issuance(), init_total_issuance - 10);
+				drop(imbalance);
+				assert_eq!(Balances::total_issuance(), init_total_issuance);
+			});
+		}
+
+		#[test]
+		fn transfer_keep_alive_works() {
+			<$ext_builder>::default().existential_deposit(1).build().execute_with(|| {
+				let _ = Balances::deposit_creating(&1, 100);
+				assert_noop!(
+					Balances::transfer_keep_alive(Some(1).into(), 2, 100),
+					Error::<$test, _>::KeepAlive
+				);
+				assert_eq!(Balances::is_dead_account(&1), false);
+				assert_eq!(Balances::total_balance(&1), 100);
+				assert_eq!(Balances::total_balance(&2), 0);
+			});
+		}
+
+		#[test]
+		#[should_panic = "the balance of any account should always be more than existential deposit."]
+		fn cannot_set_genesis_value_below_ed() {
+			($existential_deposit).with(|v| *v.borrow_mut() = 11);
+			let mut t = frame_system::GenesisConfig::default().build_storage::<$test>().unwrap();
+			let _ = GenesisConfig::<$test> {
+				balances: vec![(1, 10)],
+			}.assimilate_storage(&mut t).unwrap();
+		}
+
+		#[test]
+		fn dust_moves_between_free_and_reserved() {
+			<$ext_builder>::default()
+				.existential_deposit(100)
+				.build()
+				.execute_with(|| {
+					// Set balance to free and reserved at the existential deposit
+					assert_ok!(Balances::set_balance(RawOrigin::Root.into(), 1, 100, 0));
+					// Check balance
+					assert_eq!(Balances::free_balance(1), 100);
+					assert_eq!(Balances::reserved_balance(1), 0);
+
+					// Reserve some free balance
+					assert_ok!(Balances::reserve(&1, 50));
+					// Check balance, the account should be ok.
+					assert_eq!(Balances::free_balance(1), 50);
+					assert_eq!(Balances::reserved_balance(1), 50);
+
+					// Reserve the rest of the free balance
+					assert_ok!(Balances::reserve(&1, 50));
+					// Check balance, the account should be ok.
+					assert_eq!(Balances::free_balance(1), 0);
+					assert_eq!(Balances::reserved_balance(1), 100);
+
+					// Unreserve everything
+					Balances::unreserve(&1, 100);
+					// Check balance, all 100 should move to free_balance
+					assert_eq!(Balances::free_balance(1), 100);
+					assert_eq!(Balances::reserved_balance(1), 0);
+				});
+		}
+
+		#[test]
+		fn account_deleted_when_just_dust() {
+			<$ext_builder>::default()
+				.existential_deposit(100)
+				.build()
+				.execute_with(|| {
+					// Set balance to free and reserved at the existential deposit
+					assert_ok!(Balances::set_balance(RawOrigin::Root.into(), 1, 50, 50));
+					// Check balance
+					assert_eq!(Balances::free_balance(1), 50);
+					assert_eq!(Balances::reserved_balance(1), 50);
+
+					// Reserve some free balance
+					let _ = Balances::slash(&1, 1);
+					// The account should be dead.
+					assert!(Balances::is_dead_account(&1));
+					assert_eq!(Balances::free_balance(1), 0);
+					assert_eq!(Balances::reserved_balance(1), 0);
+				});
+		}
+
+		#[test]
+		fn account_transfer_balance_works() {
```

Comment:
This is the added test

Reply:
Rename to `account_transfer_balance_should_work`?